File: src/main/scala/org/haoxin/bigdata/flink/streaming/customSource/demo1.scala
package org.haoxin.bigdata.flink.streaming.customSource

import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
import org.apache.flink.streaming.api.windowing.time.Time

/**
  * @author huolibin@haoxin.cn
  * @date Created by sheting on 2019/7/30 17:13
  *
  */
object demo1 {
  def main(args: Array[String]): Unit = {

    val env = StreamExecutionEnvironment.getExecutionEnvironment

    //隐式转换
    import org.apache.flink.streaming.api.scala._
    val text: DataStream[Long] = env.addSource(new MyNoParallelSourceScala)
    //数据处理
    val mapData = text.map(line => {
      println("接收到的数据:" + line)
      line
    })

    val sum = mapData.timeWindowAll(Time.seconds(5)).sum(0)

    sum.print().setParallelism(1);
    env.execute("demo1")
  }

}

File: src/main/java/org/haoxin/bigdata/streaming/customSource/MysqlSource/demo1.java
package org.haoxin.bigdata.streaming.customSource.MysqlSource;

import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * @author huolibin@haoxin.cn
 * @date Created by sheting on 2019/7/17 16:18
 * 对自定义mysql source的测试
 */
public class demo1 {
    public static void main(String[] args) throws Exception {
        //获取环境
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        //获取数据
        DataStreamSource<Tuple2<String, String>> text = env.addSource(new JdbcReader()).setParallelism(1);

        text.print().setParallelism(1);
        env.execute("demo1");

    }
}
